lib/integrations/externalcloudaudit/bootstrap.go

```diff
@@ -0,0 +1,397 @@
+	_, err := clt.CreateWorkGroup(ctx, &athena.CreateWorkGroupInput{
+		Name:          &workgroup,
+		Configuration: &athenatypes.WorkGroupConfiguration{},
+	})
+	if err != nil {
+		if !strings.Contains(err.Error(), "is already created") {
+			return trace.Wrap(err, "creating athena workgroup")
+		}
+	}
+
+	return nil
+}
+
+type createGlueInfraClient interface {
+	// Creates a new database in a Data Catalog.
+	CreateDatabase(ctx context.Context, params *glue.CreateDatabaseInput, optFns ...func(*glue.Options)) (*glue.CreateDatabaseOutput, error)
+	// Creates a new table definition in the Data Catalog
+	CreateTable(ctx context.Context, params *glue.CreateTableInput, optFns ...func(*glue.Options)) (*glue.CreateTableOutput, error)
+	// Updates a metadata table in the Data Catalog.
+	UpdateTable(ctx context.Context, params *glue.UpdateTableInput, optFns ...func(*glue.Options)) (*glue.UpdateTableOutput, error)
+}
+
+// createGlueInfra creates necessary infrastructure for glue operations
+// https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsglue.html
+// Required IAM Permissions:
+// * CreateDatabase
+// * CreateTable
+// * UpdateTable
+func createGlueInfra(ctx context.Context, clt createGlueInfraClient, table, database, eventBucket string) error {
+	_, err := clt.CreateDatabase(ctx, &glue.CreateDatabaseInput{
+		DatabaseInput: &gluetypes.DatabaseInput{
+			Name:        &database,
+			Description: aws.String(databaseDescription),
+		},
+	})
+	if err != nil {
+		var aee *gluetypes.AlreadyExistsException
+		if !errors.As(err, &aee) {
+			return trace.Wrap(err, "creating glue database")
+		}
+	}
+
+	// Currently matches table input as specified in:
+	// https://github.com/gravitational/cloud/blob/22393dcc9362ec77b0a111c3cc81b65df19da0b0/pkg/tenantcontroller/athena.go#L458-L504
+	// TODO: Consolidate source of truth to a single location. Preferably teleport repository
+	// We do want to ensure that the table that exists has the correct table input so we'll update already existing tables
+	_, err = clt.CreateTable(ctx, &glue.CreateTableInput{
+		DatabaseName: &database,
+		TableInput:   getGlueTableInput(table, eventBucket),
+	})
+	if err != nil {
+		var aee *gluetypes.AlreadyExistsException
+		if !errors.As(err, &aee) {
+			return trace.Wrap(err, "creating glue table")
+		}
+
+		_, err = clt.UpdateTable(ctx, &glue.UpdateTableInput{
+			DatabaseName: &database,
+			TableInput:   getGlueTableInput(table, eventBucket),
+		})
+		if err != nil {
+			return trace.Wrap(err, "updating glue table")
+		}
+	}
+
+	return nil
+}
+
+// validateAndParseS3Input parses and checks s3 input uris against are strict rules
+// We currently enforce two buckets one for long term storage and one for transient short term storage
+func validateAndParseS3Input(input *ecatypes.ExternalCloudAuditSpec) (string, string, error) {
+	if input == nil {
+		return "", "", trace.BadParameter("input is nil")
```

Comment:
Updated Bootstrap Infra to include this instead of in validateAndParseS3Input: [ffc9570](https://github.com/gravitational/teleport/pull/33272/commits/ffc95703e2d1522c1ebfb67a951be6683b284046).